yt_idefix/data_structures.py

```diff
@@ -724,3 +816,290 @@ def _validate_units_override_keys(cls, units_override):
             )
 
         super(cls, cls)._validate_units_override_keys(units_override)
+
+
+class PlutoXdmfDataset(PlutoVtkDataset):
+    _index_class = PlutoXdmfHierarchy
+    _field_info_class = PlutoXdmfFields
+    _dataset_type = "pluto-xdmf"
+    _version_regexp = re.compile(r"\d+\.\d+\.?\d*[-\w+]*")
+    _required_header_keyword = "PLUTOXdmf"
+    _default_definitions_header = "definitions.h"
+    _default_inifile = "pluto.ini"
+    _field_offset_index = None
+
+    def _parse_parameter_file(self):
+        # IdefixDataset._parse_parameter_file()
+        grid_file = os.path.join(
+            os.path.dirname(self.parameter_filename), "grid.out"
+        )  # data-loc/grid.out
+        (
+            self.parameter_filename[:-2] + "xmf"
+        )  # data.%04d.<dbl/flt>.h5 -> data.%04d.<dbl/flt>.xmf
+        out_file = os.path.join(
+            os.path.dirname(self.parameter_filename),
+            self.parameter_filename[-6:]
+            + ".out",  # data.%04d.<dbl/flt>.h5 -> <dbl/flt>.h5.out
+        )
+        self._default_inifile = os.path.join(
+            os.path.dirname(self.parameter_filename), self._default_inifile
+        )
+        self._default_definitions_header = os.path.join(
+            os.path.dirname(self.parameter_filename), self._default_definitions_header
+        )
+        if os.path.exists(self._default_definitions_header):
+            with open(self._default_definitions_header) as deftxt:
+                lines = deftxt.readlines()
+                for line in lines:
+                    if "USER_DEF_PARAMETERS" in line:
+                        self.inputParamCount = int(line.split()[-1])
+
+        """
+        grid.out file has entries like the following:
+
+            # DIMENSIONS: 2
+            # GEOMETRY:   POLAR
+            # X1: [ 0.040000,  0.500000], 400 point(s), 2 ghosts
+            # X2: [ 0.000000,  6.283185], 400 point(s), 2 ghosts
+
+        These lines need to be parsed to create the appropriate grid data structure in yt.
+        Splitting and extracting the numers become straightforward when '[', ']' and ',' characters are removed before the split.
+        """
```

Comment:
please avoid using multiline strings as comments.
this should either be using `#` or be included in the docstring, whatever you like best

Reply:
Would you please put this as a docstring wherever you deem appropriate? I'll commit that.